pkg/networkserver/downlink.go

```diff
@@ -1203,20 +1186,24 @@ func (ns *NetworkServer) processDownlinkTask(ctx context.Context) error {
 
 				var sets []string
 				if dev.MACState.RxWindowsAvailable {
-					a := ns.attemptClassADownlink(ctx, dev, phy, fp, maxUpLength)
+					a := ns.attemptClassADataDownlink(ctx, dev, phy, fp, maxUpLength)
 					sets = append(sets, a.SetPaths...)
 					queuedEvents = append(queuedEvents, a.QueuedEvents...)
 					queuedApplicationUplinks = a.AppendApplicationUplinks(queuedApplicationUplinks...)
-					if !a.NextDownlinkAt.IsZero() {
-						nextDownlinkAt = a.NextDownlinkAt
-					}
 					if a.Scheduled {
+						t, hasNext := nextDataDownlinkAt(ctx, dev, phy, ns.defaultMACSettings)
+						if hasNext {
+							earliestAt := a.DownAt.Add(dev.MACState.CurrentParameters.Rx1Delay.Duration())
+							if t.Before(earliestAt) {
+								nextDownlinkAt = earliestAt
+							} else {
+								nextDownlinkAt = t
```

Comment:
This is intended as `nextDownlinkAt = max(t, earliestAt)` right?

Reply:
yes